### ISSUE TITLE
Godot 4.0 editor opens on wrong monitor

### ISSUE THREAD

**Issue by hhyyrylainen** [godotengine/godot#56340]
### Godot version

v4.0.dev.custom_build.91b97dac0

### System information

Linux (Fedora 35), Wayland, Vulkan API 1.2.189 - Using Vulkan Device #0: AMD - AMD RADV SIENNA_CICHLID (6900 XT)

### Issue description

When starting the Godot 4.0 editor (project manager or opening an existing project from the project list) it opens on my third monitor always. This is different from 3.4 which manages to correctly open on the monitor my cursor is positioned on. So I think there's a regression regarding multi monitor handling in 4.0.

I noticed that at least when running the game from the Godot editor *that* opens on the right monitor, but the project itself when opened from the projects list, is opened on the wrong monitor. So 2 out of 3 things open on the wrong monitor for me.

### Steps to reproduce

- Checkout the latest master code
- Compile with `scons target=release_debug -j 30 tools=yes`
- Run editor from terminal: `bin/godot.linuxbsd.opt.tools.64`
- Observe how the project manager window opens on the wrong monitor instead of the monitor I have the terminal window on and my cursor

### Minimal reproduction project

problem is with the Godot editor itself, no project is required to experience this issue

**Comment by Calinou** [comment]
I can confirm this on `master` 28174d531, regardless of single-window mode status in the Editor Settings.

I remember seeing this issue already reported elsewhere though.

**Comment by HamAndEggs** [comment]
Hi, hope you don't mind a 'me too' comment.

I am seing this issue on my system with three monitors. Godot 3.5 does not have this issue. I hope the info below will help. 

CPU: 12th Gen Intel(R) Core(TM) i5-1240P
Kernel: Linux 6.0.6-76060006-generic
OS: Pop!_OS 22.04 LTS
Compositor: Wayland
GPU Driver: i915 Mesa 22.2.0
GPU: Alder Lake-P Integrated Graphics Controller
RAM: 64GB
uname -a : Linux pop-os 6.0.6-76060006-generic #202210290932~1667401208~22.04~d2df702 SMP PREEMPT_DYNAMIC Wed N x86_64 x86_64 x86_64 GNU/Linux

Three 1080p Monitors. The Primary monitor is in the center in landscape mode. The other two flank it in portrate mode. All monitors are the same make and model.

**Comment by akien-mga** [comment]
Is this still reproducible in 4.0 RC 3 or later?

**Comment by hhyyrylainen** [comment]
I just grabbed the latest master branch commit to test. Now it looks like the new version always opens on my primary monitor, which is an improvement. 
But it still doesn't open on the monitor where my cursor is. For example opening the godot executable through a file browser on my third monitor will make the Godot window open on my primary monitor. Godot 3.5 passes that test.

**Comment by naturally-intelligent** [comment]
Yes I have this issue too, it always opens on Primary. Which isn't a huge deal, but v3 was behaving nicely. I have a script to force it to open in the monitor I'd like:

`./Godot_v4.0-stable_linux.x86_64 -e ~/path/to/project.godot --screen 1`

**Comment by Red3Tango** [comment]
This may be related. On Windows (10) with Godot v4.2.stable Editor/Project Settings (and other multi-window windows) are opening on my 3rd monitor. Would expect them on the Primary Screen or the same screen as the editor?

My setup is 3 monitors (same resolution), (3,1,2, primary in center), the setting for opening the `Editor Screen` and `Project Manager Screen` properly detects and opens on `Primary Screen` (or manual 1,2,3). Other windows however, (such as Editor and Project Settings) will open up on the 3rd screen by default (will preserve location afterwards for that project).

Enabling (then disabling) `Single Window Mode`, will revert location to 3rd monitor again. This may be expected behavior to "reset" window locations when going back to a multi-window mode.

**Comment by Etrinque** [comment]
this is as well happening to me on Widnows 11,  Godot 4.2 stable, except my 4th monitor, which is a 800X600 small utility panel inside my pc.... so not ideal.
